src/main/java/com/microsoft/azure/relay/TokenRenewer.java

```diff
@@ -61,23 +68,38 @@ void onRenewTimer() {
 	}
 
 	private void scheduleRenewTimer(SecurityToken token) {
-		this.renewTimer = new Timer();
-		Duration interval = Duration.between(Instant.now(), token.getExpiresAtUtc());
-		if (interval.isNegative()) {
-			RelayLogger.logEvent("tokenRenewNegativeDuration", this.listener);
-			return;
-		}
+	    if (this.renewTimer == null) {
+	        this.renewTimer = new Timer();    
+	    }
+	    
+	    if (scheduleLock.tryLock()) {
+	        try {
+	            Duration interval = Duration.between(Instant.now(), token.getExpiresAtUtc());
+	            if (interval.isNegative()) {
+	                RelayLogger.logEvent("tokenRenewNegativeDuration", this.listener);
+	                return;
+	            }
 
-		// TokenProvider won't return a token which is within 5min of expiring so we don't have to pad here.
-		interval = interval.compareTo(RelayConstants.CLIENT_MINIMUM_TOKEN_REFRESH_INTERVAL) < 0 ? 
-				RelayConstants.CLIENT_MINIMUM_TOKEN_REFRESH_INTERVAL : interval;
+	            // TokenProvider won't return a token which is within 5min of expiring so we don't have to pad here.
+	            interval = interval.compareTo(RelayConstants.CLIENT_MINIMUM_TOKEN_REFRESH_INTERVAL) < 0 ? 
+	                    RelayConstants.CLIENT_MINIMUM_TOKEN_REFRESH_INTERVAL : interval;
 
-		RelayLogger.logEvent("tokenRenewScheduled", this.listener, interval.toString());
-		this.renewTimer.schedule(new TimerTask() {
-			@Override
-			public void run() {
-				onRenewTimer();
-			}
-		}, interval.toMillis());
+	            if (this.renewTimerTask != null) {
+	                this.renewTimerTask.cancel();
```

Comment:
>this.renewTimerTask.cancel(); [](start = 17, length = 29)

Would you consider putting a comment for simpletons like me to understand this? Something like `// Cancel any previously scheduled task`